packages/core/src/graph/graph.ts

```diff
@@ -1783,14 +1785,52 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
       comboConfig = combo;
     }
 
-    // step2: 更新 children，根据类型添加 comboId 或 parentId
+    // step 2: Pull children out of their parents
+    let comboTrees = this.get('comboTrees');
+    const isChildById = new Set(children);
+    const pulledComboTreesById = new Map();
+
+    if (comboTrees) {
+      comboTrees.forEach(ctree => {
+        traverseTreeUp<ComboTree>(ctree, (treeNode, parentTreeNode, index) => {
+          if (isChildById.has(treeNode.id)) {
```

Comment:
Stylistic suggestion, so up to you. It might be easier to read if you reduce the number of indentations by doing this:
```ts
if (!isChildById.has(treeNode.id)) {
  return
}
...
```

Reply:
Actually... no - I think I'll keep it like this. I am a fan of early returns at the start of a function - those make total sense.

But in this case:
- `return` inside `forEach`s are kinda harder to read, since you have to keep reminding yourself that those are actually `continue`s in practice
- I'd have to use `return true`, where the value has a specific meaning (don't interrupt traversal)
- I'd have 2 return instructions like this... not the best